test/sql/alter/test_rename_table.cpp

```diff
@@ -0,0 +1,108 @@
+#include "catch.hpp"
+#include "test_helpers.hpp"
+
+using namespace duckdb;
+using namespace std;
+
+TEST_CASE("Test RENAME TABLE single transaction", "[alter]") {
+    DuckDB db(nullptr);
+    Connection con(db);
+    REQUIRE_NO_FAIL(con.Query("CREATE TABLE tbl(i INTEGER)"));
+    REQUIRE_NO_FAIL(con.Query("BEGIN TRANSACTION"));
+    REQUIRE_NO_FAIL(con.Query("ALTER TABLE tbl RENAME TO tbl2"));
+    REQUIRE_NO_FAIL(con.Query("SELECT * FROM tbl2"));
+    REQUIRE_FAIL(con.Query("SELECT * FROM tbl"));
+    REQUIRE_NO_FAIL(con.Query("ROLLBACK"));
+    REQUIRE_NO_FAIL(con.Query("SELECT * FROM tbl;"));
+    REQUIRE_FAIL(con.Query("SELECT * FROM tbl2"));
+    REQUIRE_NO_FAIL(con.Query("BEGIN TRANSACTION;"));
+    REQUIRE_NO_FAIL(con.Query("ALTER TABLE tbl RENAME TO tbl2"));
+    REQUIRE_NO_FAIL(con.Query("COMMIT"));
+    REQUIRE_NO_FAIL(con.Query("SELECT * FROM tbl2"));
+    REQUIRE_FAIL(con.Query("SELECT * FROM tbl"));
+}
+
+TEST_CASE("Test RENAME TABLE two parallel transactions", "[alter]") {
+    DuckDB db(nullptr);
+    Connection con1(db);
+    Connection con2(db);
+    REQUIRE_NO_FAIL(con1.Query("CREATE TABLE tbl(i INTEGER)"));
+    REQUIRE_NO_FAIL(con1.Query("INSERT INTO tbl VALUES (3)"));
+
+    REQUIRE_NO_FAIL(con1.Query("BEGIN TRANSACTION"));
+    REQUIRE_NO_FAIL(con2.Query("BEGIN TRANSACTION"));
+    REQUIRE_NO_FAIL(con1.Query("ALTER TABLE tbl RENAME TO tbl2"));
+
+    REQUIRE_NO_FAIL(con1.Query("SELECT * FROM tbl2"));
```

Comment:
I'm missing the result check here: this test should also verify that no data is lost when a table is renamed (by using `CHECK_COLUMN`).

Reply:
I will add that.